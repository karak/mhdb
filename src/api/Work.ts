/**
 * Work of haiku
 */
export default interface Work {
  /** ID */
  id: string;
  /** Body */
  body: string;
  /** Author */
  author: string;
};
