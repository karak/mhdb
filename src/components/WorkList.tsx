import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { List, ListItem } from 'material-ui/List';
import Work from '../api/Work';

export interface WorkListProps {
  items?: ReadonlyArray<Work>;
}

// tslint:disable-next-line:function-name
export default function WorkList(props: WorkListProps) {
  return (
    <List>
      {(props.items || []).map(x => (
        <ListItem
          key={x.id}
          primaryText={x.body}
          secondaryText={x.author}
          secondaryTextLines={1}
        />
      ))}
    </List>
  );
}
