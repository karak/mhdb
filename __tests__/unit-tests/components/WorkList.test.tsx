import '../../helpers/setup';
import * as React from 'react';
import WorkList from '../../../src/components/WorkList';
import Work from '../../../src/api/Work';
import { shallow } from 'enzyme';
import { default as toJson } from 'enzyme-to-json';

describe('ErrorIcon', () => {
  it('matches with snapshot', () => {
    const items: Work[] = [
      {
        id: 'work-1',
        author: 'John',
        body: 'ABC',
      },
      {
        id: 'work-2',
        author: 'Mary',
        body: 'XYZ',
      },
    ];
    const tree = toJson(shallow(<WorkList items={items} />));

    expect(tree).toMatchSnapshot();
  });
});
