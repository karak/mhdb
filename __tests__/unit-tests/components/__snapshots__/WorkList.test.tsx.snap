// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorIcon matches with snapshot 1`] = `
<List>
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    key="work-1"
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="ABC"
    primaryTogglesNestedList={false}
    secondaryText="John"
    secondaryTextLines={1}
  />
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    key="work-2"
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="XYZ"
    primaryTogglesNestedList={false}
    secondaryText="Mary"
    secondaryTextLines={1}
  />
</List>
`;
